src/views/VallisMap.vue

```diff
@@ -159,175 +121,192 @@ function onEachOddity(feature, layer) {
   layer.bindPopup(popup.$mount().$el, { minWidth: 320 });
 }
 
+function toroidMarkerFromName(name) {
+  if (name.startsWith('Calda')) {
+    return caldaMarker;
+  } else if (name.startsWith('Sola')) {
+    return solaMarker;
+  } else if (name.startsWith('Vega')) {
+    return vegaMarker;
+  }
+
+  return null;
+}
+
+function caveMarkerFromName(name) {
+  if (name.startsWith('Fishing Cave')) {
+    return fishCaveMarker;
+  } else if (name.startsWith('Toroid Cave')) {
+    return toroidCaveMarker;
+  } else if (name.startsWith('Toroid Fish Cave')) {
+    return fishToroidCaveMarker;
+  }
+
+  return null;
+}
+
+function defaultToggleValues() {
+  let defaults = {};
+  dataFn().geo.forEach((g) => {
+    defaults[g.name + '-toggle-value'] = true;
+  });
+  return defaults;
+}
+
 const markerAlias = L.marker;
 const labelAlias = L.circleMarker;
 
-export default {
-  name: 'Vallismap',
-  data() {
-    return {
-      zoom: -1,
-      center: L.latLng(942, 1060),
-      url: vallis,
-      bounds: [[0, 0], [2150, 2153]],
-      mapOptions: {
-        zoomSnap: 0.5,
-        minZoom: -10,
-        attributionControl: false,
-      },
-      crs: L.CRS.Simple,
-      mapStyle: {
-        height: 'calc(100vh - 100px)',
-        width: '100%',
-      },
-      geo: [
-        {
-          name: 'Map Label',
-          json: labels,
-          opts: {
-            pointToLayer: function(feature, latlng) {
-              return labelAlias(latlng)
-                .setStyle({
-                  stroke: false,
-                  fill: false,
-                })
-                .bindTooltip(feature.properties.name, {
-                  permanent: true,
-                  direction: 'center',
-                  className: 'map-label',
-                })
-                .openTooltip();
-            },
-          },
-        },
-        {
-          name: 'Fishing',
-          json: fish,
-          opts: {
-            pointToLayer: function(feature, latlng) {
-              return markerAlias(latlng);
-            },
-            onEachFeature: onEachFeature,
-          },
-        },
-        {
-          name: 'Fishing Spots',
-          json: fishRecommend,
-          opts: {
-            pointToLayer: function(feature, latlng) {
-              return markerAlias(latlng, { icon: fishRecommendMarker });
-            },
-            onEachFeature: onEachFeature,
+function dataFn() {
```

Comment:
not a big fan of the naming

Reply:
Would getData() be better?

Reply:
either getData or what's there now (`data`) is fine